File: Source/ufw/Source/UAPPTemplate/UAppFrameworkTemplate/Philips uApp.xctemplate/___PROJECTNAMEASIDENTIFIER___Interface.m
//
//  ___FILENAME___
//  ___PACKAGENAME___
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//___COPYRIGHT___
//

#import "___PROJECTNAMEASIDENTIFIER___Interface.h"
#import <UIKit/UIKit.h>

@implementation ___PROJECTNAMEASIDENTIFIER___Interface

- (instancetype)initWithDependencies:(UAPPDependencies * _Nonnull) dependencies andSettings : (UAPPSettings * _Nullable) settings {
    <#Initialise your uApp here#>
}

- (UIViewController * _Nullable)instantiateViewController:(UAPPLaunchInput * _Nonnull) launchInput withErrorHandler:(void (^ _Nullable)(NSError * _Nullable error))completionHandler {
    
    UIStoryboard *storyBoard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
    UIViewController *viewController = [storyBoard instantiateViewControllerWithIdentifier:@"___PROJECTNAMEASIDENTIFIER___ViewController"];
   
    return viewController;
}

@end

File: Source/ufw/Source/UAPPTemplate/UAppFrameworkTemplate/Philips uApp.xctemplate/___PROJECTNAMEASIDENTIFIER___Interface.swift
//
//  ___FILENAME___
//  ___PACKAGENAME___
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//___COPYRIGHT___
//

import UIKit
import UAPPFramework


class ___PROJECTNAMEASIDENTIFIER___Interface: NSObject, UAPPProtocol {
    private var appDependencies : ___PROJECTNAMEASIDENTIFIER___Dependencies!
    
    private var launchInputs : ___PROJECTNAMEASIDENTIFIER___LaunchInput?
    
    required init(dependencies: UAPPDependencies, andSettings settings: UAPPSettings?) {
        super.init()
        self.appDependencies = dependencies as! ___PROJECTNAMEASIDENTIFIER___Dependencies
    }
    
    func instantiateViewController(_ launchInput: UAPPLaunchInput, withErrorHandler completionHandler: ((Error?) -> Void)? = nil) -> UIViewController? {
        
        
        let storyBoard = UIStoryboard(name:"Main", bundle: nil)
        let viewController = storyBoard.instantiateViewControllerWithIdentifier("___PROJECTNAMEASIDENTIFIER___ViewController")
        
        return viewController
    }
}
